src/docbuilder/rustwide_builder.rs

```diff
@@ -383,16 +383,19 @@ impl RustwideBuilder {
         let successful = build_dir
             .build(&self.toolchain, &krate, self.prepare_sandbox(&limits))
             .run(|build| {
-                (|| -> Result<bool> {
-                    use docsrs_metadata::BuildTargets;
+                let metadata = Metadata::from_crate_root(build.host_source_dir())?;
+                let BuildTargets {
+                    default_target,
+                    other_targets,
+                } = metadata.targets(self.config.include_default_targets);
+                let mut targets = vec![default_target];
+                targets.extend(&other_targets);
+                // Fetch this before we enter the sandbox, so networking isn't blocked.
+                build.fetch_build_std_dependencies(&targets)?;
```

Comment:
Since this is run for every build, will this be a no-op if the dependencies are already fetched?

Reply:
yup, exactly :)

```
; cargo fetch -Zbuild-std --target aarch64-apple-darwin
    Updating crates.io index
    Updating crates.io index
  Downloaded cfg-if v1.0.0
  Downloaded gimli v0.26.2
  Downloaded unicode-width v0.1.10
  Downloaded miniz_oxide v0.5.3
  Downloaded rustc-demangle v0.1.21
  Downloaded libc v0.2.140
  Downloaded adler v1.0.2
  Downloaded object v0.29.0
  Downloaded getopts v0.2.21
  Downloaded compiler_builtins v0.1.91
  Downloaded cc v1.0.77
  Downloaded addr2line v0.17.0
  Downloaded memchr v2.5.0
  Downloaded hashbrown v0.12.3
  Downloaded 14 crates (2.2 MB) in 0.51s
  Downloaded syn v2.0.15
  Downloaded async-trait v0.1.68
  Downloaded proc-macro2 v1.0.56
  Downloaded quote v1.0.26
  Downloaded unicode-ident v1.0.8
  Downloaded 5 crates (384.2 KB) in 0.25s
; cargo fetch -Zbuild-std --target aarch64-apple-darwin
```